egs/tedlium2/align1/run.sh

```diff
@@ -0,0 +1,194 @@
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+. ./path.sh || exit 1;
+. ./cmd.sh || exit 1;
+
+# general configuration
+python=python3
+backend=pytorch
+stage=-1       # start from -1 if you need to start from data download
+stop_stage=100
+ngpu=1         # number of gpus ("0" uses cpu, otherwise use gpu)
+debugmode=1
+dumpdir=dump   # directory to dump full features
+N=0            # number of minibatches to be used (mainly for debugging). "0" uses all minibatches.
+verbose=1      # verbose option
+resume=        # Resume the training from snapshot
+
+# feature configuration
+do_delta=false
+cmvn=
+
+preprocess_config=conf/specaug.yaml
+train_config=conf/train.yaml
+lm_config=conf/lm.yaml
+decode_config=conf/decode.yaml
+
+# rnnlm related
+lm_resume=        # specify a snapshot file to resume LM training
+lmtag=            # tag for managing LMs
+
+# decoding parameter
+recog_model=model.acc.best # set a model to be used for decoding: 'model.acc.best' or 'model.loss.best'
+n_average=10
+
+# bpemode (unigram or bpe)
+nbpe=500
+bpemode=unigram
+
+# exp tag
+tag="" # tag for managing experiments.
+
+. utils/parse_options.sh || exit 1;
+
+# Set bash to 'debug' mode, it will exit on :
+# -e 'error', -u 'undefined variable', -o ... 'error in pipeline', -x 'print commands',
+set -e
+set -u
+set -o pipefail
+
+train_set=train_trim_sp
+train_dev=dev_trim
+recog_set="dev test"
+models=tedlium2.rnn.v2
+
+# Parameters for CTC alignment
+# The subsampling factor depends on whether the encoder uses subsampling
+subsampling_factor=4
+# minium confidence score in log space - may need adjustment depending on data and model, e.g. -1.5 or -5.0
+min_confidence_score=-5.0
+
+download_dir=models
+align_model=
+align_config=
+align_dir=align
+api=v1
+dict=
+
+if [ ${stage} -le -2 ] && [ ${stop_stage} -ge -2 ]; then
+    echo "stage -1: Data Download"
+    local/download_data.sh
+fi
+
+if [ ${stage} -le -1 ] && [ ${stop_stage} -ge -1 ]; then
+    echo "stage -1: Model Download"
+    . local/download_model.sh
+fi
+
+
+# Download trained models
+if [ -z "${cmvn}" ]; then
+    cmvn=$(find ${download_dir}/${models} -name "cmvn.ark" | head -n 1)
+fi
+if [ -z "${align_model}" ]; then
+    align_model=$(find ${download_dir}/${models} -name "model*.best*" | head -n 1)
+fi
+if [ -z "${align_config}" ]; then
+    align_config=$(find ${download_dir}/${models} -name "decode*.yaml" | head -n 1)
+fi
+if [ -z "${dict}" ]; then
+    dict=$(find ${download_dir}/${models}/data/lang_*char -name "*.txt" | head -n 1) || dict=""
+
+    if [ -z "${dict}" ]; then
+        mkdir ${download_dir}/${models}/data/lang_autochar/ -p
+        model_config=$(find ${download_dir}/${models}/exp/*/results/model.json | head -n 1)
+        cat $model_config | python -c 'import json,sys;obj=json.load(sys.stdin);[print(char + " " + str(i + 1)) for i, char in enumerate(obj[2]["char_list"])]' > ${download_dir}/${models}/data/lang_autochar/dict.txt
```

Comment:
@sw005320 I wonder if `python` command uses python in which the ESPnet is installed. Is it correct to use the `python` command to run python scripts in the ESPnet environment?

Reply:
I think this would be OK.

Reply:
Changed `python` to `${python}` in this case.